src/test/java/seedu/intrack/model/internship/TaskTest.java

```diff
@@ -0,0 +1,32 @@
+package seedu.intrack.model.internship;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.intrack.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class TaskTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Task(null, null));
+    }
+
+    @Test
+    public void constructor_invalidTask_throwsIllegalArgumentException() {
+        String invalidTaskName = "";
+        String invalidTaskTime = "TODAY";
+        assertThrows(IllegalArgumentException.class, () -> new Task(invalidTaskName, invalidTaskTime));
+    }
```

Comment:
Still on the idea same idea as above, would it be better to separate invalid inputs for taskName and taskTime?

Reply:
Thank you for pointing that out, I will update the test case now.